### DISCUSSION TITLE
How to access camera tool photos?

### DISCUSSION THREAD

**Discussion by ju57m3** [mozilla/hubs#2950] [General Questions & Discussion] [answered]
hi!

i'm running hubs cloud on aws and would like to access the photos users create with the camera tool ( i want to use them in a unity3d environment for a projection project).

while i can access them via a link (eg.: https://[server-adress]/files/0b9228c1-b987-4c74-b7d3-1dbf67d8c47d.png), i can only find the corresponding .blob and .meta.json files on the server in some two digit named subfolder inside storage/owned.

i am obviously missing some basics here, unfortunately googling blobs raised more questions than it answered.

does anyone know how to access the "files" folder in the link above, respectively how to get a list of file links i can then use to download them?

thanks in advance

**Comment by emclaren** [comment, marked as answer]
Hi, 

with the way that Hubs is currently set up this isn't easily possible. I spoke to my colleagues and received this information that might be of use: 

The images are encrypted on the database/s3 bucket. The admin can't access the user's files, if they weren't in the room to get the key.  You will only find the blobs because that's how they are encrypted them so they can only be accessed w/ the correct key

Note that anything persistent, though still encrypted, also has the keys stored

We have 2 concepts for files:

"Unowned" temporary files that get automatically deleted after 72 hours. All files start in this state and are not associated with any account. You need to provide the "encryption token" as part of the request to view the file as it is not stored anywhere on our servers.

"Owned" persistent files that are always tied to a specific account and always come by "promoting" an "Unowned" file by providing the encryption token and a "promotion token" that you got when making the "unowned" file. The keys for these files are stored in the database and can be requested without providing a key (they are still encrypted on disk).

There is no API for directly "promoting" a file, instead this happens as part of other requests that need persistent files to exist. Like uploading an avatar, creating a scene, pinning an object, etc.

Because there is no mechanism to turn off encryption for Hubs Cloud photos, another option is to connect the Hubs Discord Bot to a Discord server you own. The Hubs Bot lets you bridge chat & media created in the room to Discord. This way you could see links to all the media created by users in the space. Note that the links to the media would only work for 72 hours so you would need to be regularly opening and saving them. Setting up the Discord Bot on Hubs Cloud is currently a bit more difficult than with hubs.mozilla.com, but we are working on making it easier.

**Reply by ju57m3** [reply]
Hi!

Thank you very much, this definitely makes things clearer!

The workaround i came up with:
A cronjob on the ec2 instance that searches the owned folder for json files, creates a text file with their filenames and then syncs the text file and json files to a s3 bucket.
In unity3d i then read the filenames from the text file in the s3 bucket, check every json file for "content_type":"image/png" and download the images from https://[server-adress]/files/[json-file-name-without-extension].png

This may not be the best way, but it gets the photos users take and pin into unity3d.

I also tried getting a connection to the database, but that didn't work out and i think that's intended.

Thanks!
